gwdetchar/io/datafind.py

```diff
@@ -70,6 +75,47 @@ def check_flag(flag, gpstime, duration, pad):
     return True
 
 
+def remove_missing_channels(channels, gwfcache):
+    """Find and remove channels from a given list that are not available in
+    a given cache of frame files
+
+    Parameters
+    ----------
+    channels : `list` of `str`
+        list of requested channels
+
+    gwfcache : `list` of `str`
+        list of paths to .gwf files
+
+    Returns
+    -------
+    keep : `list` of `str`
+        list of common channels found in every file in the cache
+
+    Notes
+    -----
+    For every channel requested that is not available in `gwfcache`, a
+    `UserWarning` will be raised.
+
+    See Also
+    --------
+    gwpy.io.gwf.iter_channel_names
+        for the utility used to identify frame contents
+    """
+    # get available channels from the first and last frame file
+    available = set(io_gwf.iter_channel_names(gwfcache[0]))
+    if len(gwfcache) > 1:
+        available.update(io_gwf.iter_channel_names(gwfcache[-1]))
+    # work out which channels to keep, and which to reject
+    keep = list(set(channels) & available)
+    reject = list(set(channels) - set(keep))
```

Comment:
It would be more efficient to do something like this:

```python
    channels = set(channels)
    keep = channels & available
    reject = channels - keep
```

and then just `return list(keep)` at the end.

Reply:
Yep, done in the latest commit